_enumerable.rb

```diff
@@ -0,0 +1,84 @@
+module Enumerable
+  def my_each
+    each do |ele|
+      yield ele
+    end
+    self
+  end
+
+  def my_each_with_index
+    length.times do |indx|
+      yield self[indx], indx
+    end
+    self
+  end
+
+  def my_select
+    selected_elements = []
+    my_each do |k|
+      selected_elements.push(k) if yield k
+    end
+    selected_elements
+  end
+
+  def my_all?
+    my_each do |ele|
+      # returns false if an element that doesn't satsify the condition is found
+      return false unless yield ele
+    end
+    true
+  end
+
+  def my_any?
+    my_each do |ele|
+      return true if yield ele
+    end
+    false
+  end
```

Comment:
Is expected that:

#my_any? when no block or argument is given returns true if at least one of the collection is not false or nil

#my_any? when no block or argument is given returns false if at least one of the collection is not true

#my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class::Numeric

#my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class::Integer

#my_any? when a Regex is passed as an argument returns true if any of the collection matches the Regex

#my_any? when a Regex is passed as an argument returns false if none of the collection matches the Regex

#my_any? when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern

#my_any? when a pattern other than Regex or a Class is given returns true if any of the collection matches the pattern